File: python/razao.py
def plusMinus(arr):
    positivos = negativos = zero = 0
    for i in arr:
        if i > 0:
            positivos += 1
        if i < 0:
            negativos += 1
        if i == 0:
            zero += 1
    div_posi = positivos / len(arr)
    div_negativos = negativos / len(arr)
    div_zero = zero / len(arr)
    print(f'{div_posi:.6f}')
    print(f'{div_negativos:.6f}')
    print(f'{div_zero:.6f}')


lista = list()
tam = int(input())
for _ in range(tam):
    lista.append(int(input()))
plusMinus(lista)

File: C/razao.c
#include <stdio.h>

void plusMinus(int *arr, int t);

int main()
{
    int tam, i;
    scanf("%d",&tam);
    int vet[tam];
    for(i = 0; i < tam; i++)
    {
        scanf("%d",&vet[i]);
    }
    plusMinus(vet, tam);
}

void plusMinus(int *arr, int t)
{
    int i, positivos = 0, negativos = 0, zero = 0;
    double div_posi, div_nega, div_zero;
    for(i = 0; i < t; i++)
    {
        if(arr[i] < 0)
            negativos++;
        if(arr[i] > 0)
            positivos++;
        if(arr[i] == 0)
            zero++;
    }
    div_posi = (double) positivos / t;
    div_nega = (double) negativos / t;
    div_zero = (double) zero / t;
    printf("%.6f\n",div_posi);
    printf("%.6f\n",div_nega);
    printf("%.6f\n",div_zero);
}